File: python/ex16.py
def mult(x, y):
  mult = 0
  while(y>0):
    mult += x
    y -= 1
  return mult
def pot(a, b):
  pot = 1
  while(b>0):
    pot = mult(pot, a)
    b -= 1
  return pot
a = int(input())
b = int(input())
print(pot(a,b))

File: c/ex16.c
//Cálculo de potência usando somas
#include<stdio.h>
int mult(int a, int b){
    int soma = 0;
    if(b > a){
        int aux;
        aux = b;
        b = a;
        a = aux;
    }
    while(b > 0){
        soma += a;
        b--;
    }
    return soma;
}
int main(){
    int a, b, potencia = 1;
    scanf(" %i %i", &a, &b);
    while(b > 0){
        potencia = mult(potencia, a);
        b--;
    }
    printf("%i", potencia);
}